src/query_parser/v2/parser.y

```diff
@@ -1011,6 +989,49 @@ vector_attribute_list(A) ::= vector_attribute(B). {
   A.needResolve = array_append(A.needResolve, B.needResolve);
 }
 
+/*** Vector range queries ***/
+expr(A) ::= modifier(B) COLON LSQB vector_range_query(C) RSQB. { // top-level parse
+    C->vn.vq->property = rm_strndup(B.s, B.len);
+    if (C->vn.vq->scoreField == NULL) {
+        RedisModule_Assert(-1 != (rm_asprintf(&C->vn.vq->scoreField, "__%.*s_score", B.len, B.s)));
+    }
+    A = C;
+}
+
+vector_range_query(A) ::= vector_range_command(B). {
+    A = B;
+}
+
+vector_range_query(A) ::= vector_range_command(B) vector_attribute_list(C). {
+  B->vn.vq->params = C;
+  A = B;
+}
+
+vector_range_query(A) ::= vector_range_command(B) vector_score_field(C). {
+  B->params = array_grow(B->params, 1);
+  memset(&array_tail(B->params), 0, sizeof(*B->params));
+  QueryNode_SetParam(ctx, &(array_tail(B->params)), &(B->vn.vq->scoreField), NULL, &C);
+  A = B;
+}
+
+vector_range_query(A) ::= vector_range_command(B) vector_attribute_list(C) vector_score_field(D). {
+  B->params = array_grow(B->params, 1);
+  memset(&array_tail(B->params), 0, sizeof(*B->params));
+  QueryNode_SetParam(ctx, &(array_tail(B->params)), &(B->vn.vq->scoreField), NULL, &D);
+  B->vn.vq->params = C;
+  A = B;
+}
+
+vector_range_command(A) ::= TERM(T) param_num(B) ATTRIBUTE(C). {
+  if (strncasecmp("RANGE", T.s, T.len)) {
```

Comment:
I understand. Do you think that perhaps we could use RANGE for all types? and the parser will be able to differentiate?

Reply:
Having the same keyword for all types, is equivalent to not having any keyword - this is OK as long as the number of tokens is 2 for numeric, 3 for vector and 4 for geo, and if we would like to change that, this would cause conflicts, so a distinct keyword is probably required.